modules/cas_cache/layer_cache_management.c

```diff
@@ -1422,6 +1417,15 @@ int cache_mngt_remove_core_from_cache(struct kcas_remove_core *cmd)
 	if (result)
 		return result;
 
+	result = _cache_mngt_remove_core_is_active(cache, cmd->core_id);
+	if (result) {
+		printk(KERN_WARNING OCF_PREFIX_SHORT
+				"Core is already inactive. To manage the "
```

Comment:
This is "time of check vs time of use" kind of error. After you check whether core is active, its state might change. You should not release the management lock between checking core active state and removing the core. This means you should do both under one management write lock

Reply:
That's a very important advice. Thank you.